scrapy_poet/api.py

```diff
@@ -133,3 +134,14 @@ def parse(*args, item: page_or_item_cls, **kwargs):  # type:ignore
 
     setattr(parse, _CALLBACK_FOR_MARKER, True)
     return parse
+
+
+@dataclass
+class AnnotatedResult:
```

Comment:
If we're exposing it in `__init__`, we should probably add a docstring to this class.